admin/services/prepare-check.service.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const moment = require('moment')
+const R = require('ramda')
+const redisService = require('./redis-cache.service')
+
+const service = {
+  /**
+   * @param {[{object}]} checks the pupil checks to prepare
+   * @returns {Promise<void>}
+   */
+  prepareChecks: (checks) => {
+    if (!Array.isArray(checks)) {
+      throw new Error('checks is not an array')
+    }
+    const preparedChecks = checks.map(check => {
+      const preparedCheck = constructPreparedCheck(check)
+      return {
+        key: buildKey(check.schoolPin, check.pupilPin),
+        value: preparedCheck,
+        ttl: secondsBetweenNowAndPinExpiryTime(preparedCheck.pinExpiresAt)
+      }
+    })
+    return redisService.setMany(preparedChecks)
+  }
+}
+
+/**
+ *
+ * @param {Date} pinExpiry
+ * @returns {number} number of seconds between now and the pin expiry time
+ */
+function secondsBetweenNowAndPinExpiryTime (pinExpiry) {
```

Comment:
If pinExpiry is coming from another object it is very likely a Moment.moment object